### PR TITLE
 * disk plugin: call ignorelist_match() once per device, not for each send metric

### PR DESCRIPTION
**Pull request by rpv-tomsk**
There is no need to call ignorelist_match() for each device many times (for each collected metric). One call per cycle is enough.

### CODE REVIEW

**Comment by rubenk** [comment]
Thanks @rpv-tomsk. Can you rebase so I can merge this?

File: src/disk.c [hunk 1]
```diff
@@ -972,6 +982,12 @@ static int disk_read (void)
 	for (counter=0; counter < disks; counter++) {
 		strncpy(name, ds->disk_name, sizeof(name));
 		name[sizeof(name)-1] = '\0'; /* strncpy doesn't terminate longer strings */
+
+		if (ignorelist_match (ignorelist, name) != 0) {
```
Comment (rpv-tomsk) [inline src/disk.c]: Ruben, please look for this carefully.

File: src/disk.c [hunk 2]
```diff
@@ -510,6 +500,15 @@ static int disk_read (void)
 		else
 			ssnprintf (disk_name, sizeof (disk_name), "%i-%i", disk_major, disk_minor);
 
+		DEBUG ("disk plugin: disk_name = \"%s\"", disk_name);
+
+		/* check the name against ignore list */
+		if (ignorelist_match (ignorelist, disk_name) != 0) {
```
Comment (rpv-tomsk) [inline src/disk.c]: Ruben, please look for this carefully.

**Comment by rpv-tomsk** [comment]
Rebased.

Also I reviewed this patch again and made two additional fixes.
Please look into comments of the patch.

File: src/disk.c (hunk 1 above)
Comment (rubenk) [inline src/disk.c]: I don't understand what you mean.

File: src/disk.c (hunk 2 above)
Comment (rubenk) [inline src/disk.c]: What am I looking for?

File: src/disk.c (hunk 1 above)
Comment (rpv-tomsk) [inline src/disk.c]: There was no these changes in first PR.

File: src/disk.c (hunk 2 above)
Comment (rpv-tomsk) [inline src/disk.c]: There was no these changes in first PR, so I decided to inform you about new changes appeared.

Comment (rubenk) [inline src/disk.c]: I see, much appreciated :)

**Pull request merged**